rcgen/src/csr.rs

```diff
@@ -115,6 +115,78 @@ impl CertificateSigningRequestParams {
 								.push(SanType::try_from_general(name)?);
 						}
 					},
+					x509_parser::extensions::ParsedExtension::ExtendedKeyUsage(eku) => {
+						if eku.any {
```

Comment:
This is pretty verbose. Suggest adding a private method `fn insert_extended_key_usage(&mut self, eku: ExtendedKeyUsagePurpose)` to `CertificateParams` (at the bottom of the `impl`) that does the `contains()`/`push()` dance.